internal/host/transition_test.go

```diff
@@ -5988,3 +5989,153 @@ func validateEqualProgress(p1, p2 *models.HostProgressInfo) {
 		Expect(time.Time(p1.StageUpdatedAt).Equal(time.Time(p2.StageUpdatedAt))).To(BeTrue())
 	}
 }
+
+type testState struct {
+	state string
+}
+
+func newTestState(state string) *testState {
+	return &testState{
+		state: state,
+	}
+}
+
+func (ts *testState) State() stateswitch.State {
+	return stateswitch.State(ts.state)
+}
+
+func (ts *testState) SetState(state stateswitch.State) error {
+	ts.state = string(state)
+	return nil
+}
+
+var _ = Describe("State machine test", func() {
```

Comment:
Suggesting to have several different tests cases here. 
For example this describe can scope only refresh functions. 
Refresh is not the only operations of the state machine.